src/librustdoc/visit_ast.rs

```diff
@@ -243,13 +243,15 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
     ) {
         debug!("visiting item {:?}", item);
 
-        let receiver_kind = if let hir::ItemKind::Use(..) = item.kind {
-            AttrReceiverKind::Use
+        if let hir::ItemKind::Use(path, _) = item.kind {
+            if let hir::def::Res::Def(hir::def::DefKind::Ctor(..), _) = path.res {
+                // ignore auto-generated constructors to avoid duplicated errors
+            } else {
+                self.check_item_attrs(&item.attrs, AttrReceiverKind::Use)
+            }
```

Comment:
```suggestion
            // ignore auto-generated constructors to avoid duplicated errors
            if !matches(path.res, hir::def::Res::Def(hir::def::DefKind::Ctor(..), _)) {
                self.check_item_attrs(&item.attrs, AttrReceiverKind::Use)
            }
```

Reply:
Oh, right, this is so much cleaner :)